### ISSUE TITLE
User confusion in the neighborhood view; make note of indirect and direct relations

### ISSUE THREAD

**Issue by ValWood** [geneontology/amigo#360]
Whilst

 PomBase were looking at this (mainly to find a way to display neighbouring terms in Canto), we thought this looked a little odd.

![amigo_tracker](https://cloud.githubusercontent.com/assets/7359272/16188942/e5ce3456-36cf-11e6-9c55-6abd9711b121.jpg)

**Comment by ValWood** [comment]
We were wondering why not display as a simplere list. Currently  the term is in the heading
"Parents of detection of stimulus...."
and than repeated on every row....

It might be better to make the therm name more prominent at the top, and then not repeat.

Also the labels 'subject' and 'object' are a bit strange in this context.

Finally I'm not sure now exactly what this is meant to show. At first I thought this was only the 'immediate' parents, but for this term this doesn't appear to be the case. Will demonstrate....

**Comment by ValWood** [comment]
The neighbourhood terms are highlighted, what doe this mean?


![neighbourhood](https://cloud.githubusercontent.com/assets/7359272/16189203/0d3d163c-36d1-11e6-8ff3-a3aee62c26df.jpg)

**Comment by cmungall** [comment]
Not following the connection between these.

 - for the first, I agree the table may be overkill and a simple list may be better
 - no idea why the quickgo screenshot has things colored. how did you get here?

**Comment by ValWood** [comment]
I  added the terms  in the "neighbourhood" to the basket and viewed them in the ancestor chart.

I can't figure out how you decide what is "neighbourhood'. For example why 
'detection of stimulus'
and
'mitotic cell cycle checkpoint' 
they aren't direct ancestors....

**Comment by kltm** [comment]
@ValWood could you give the URL for the AmiGO instance that you're using, just so we're sure what version that we're looking at?
Since the default image that we present in [Graph Views] comes from QuickGO, we're not always sure the exact methods they use to generate it. The internal AmiGO graph builder is also available there, but it must be selected separately (it's a link at the bottom of the [Graph Views] tab section).
Also, what is the "cart" that you're referring to in this case? AmiGO did at one point have have some cart functionality, mostly for use with Galaxy, but that should mostly be removed from common interfaces. If you could give an example of its use...

**Comment by ValWood** [comment]
I *thought* we were using the live version, as this is what I have bookmarked 
http://amigo.geneontology.org/amigo/term/GO:0072442

However, the "neighbourhood"  tab does not appear to be present in the live version?
Maybe I took this screenshot from a screen share.....
@mah11 @kimrutherford  which AmiGO were we using? Can you remember?

(Seth, I  don't have a problem with  the QuickGO view/image. I just used QuickGO to highlight the terms which were displayed under 'neighbourhood' so I could explain that I did not understand what constituted "neighbourhood".)

**Comment by kltm** [comment]
Possible places to have seen the [Neighborhood] tab are:

* http://tomodachi.berkeleybop.org/amigo/term/GO:0072442 (labs version to go live soon)
* http://ec2-23-22-182-58.compute-1.amazonaws.com:9999/amigo/term/GO:0072442 (your fypo instance)

A bit of the discussion around the "full" neighborhood (all edges) display can be found here #220.

**Comment by ValWood** [comment]
OK so Neighbourhood is new. What determines why the terms which are not close ancestors are displayed? I still can't figure this out....

**Comment by kltm** [comment]
The use case for [Neighborhood] is that it would contain *all* direct links between two terms, not just the main ones that are typically used (something desired by the Planteome people). To be explicit and help see what might be going on, could you provide:

* the term IDs for the two terms that are under consideration
* the full URL for the page where you're seeing them in the [Neighborhood] tab

**Comment by cmungall** [comment]
Yes, this isn't just useful for planteome. In GO, it's useful to see direct links over relationships that would not be shown in the standard graph or hierarchical views. This includes for example has-part, process to chemical or cell links, etc.

The full discussion is here: #220

**Comment by ValWood** [comment]
OK, so it should display ALL direct links.  When you say "all direct links" do you mean all the terms it is related to, or just the immediate ancestors? The screen shot above (which I still don't remember where it was from), does not display  either of these,  it displays the 5 terms I highlighted in the QuickGO  graph.
(I'm not disputing its usefulness. I'm just trying to understand what it is displaying).

**Comment by cmungall** [comment]
Slightly dodging the question here, but part of the issue is that there are conflicting definitions of 'related to' and 'immediate ancestors'. My action item is to clarify these and document the behavior.

It may be current behavior is correct by some definition but suboptimal from a user perspective. For example, 'detection of stimulus involved in X' is always directly "connected" to 'detection of stimulus' by an axiom. But what the user wants to see is direct parents in the relaxed inferred graph

**Comment by ValWood** [comment]
OK when you have documented can you let me know and I'll see if I understand  this then? At present I'm lost.....

**Comment by kltm** [comment]
@cmungall Has there been any activity on this?

**Comment by cmungall** [comment]
perhaps we can close this and open distinct tickets.

The apparent redundancy issue is addressed here: https://github.com/geneontology/amigo/issues/366

**Comment by kltm** [comment]
On advice of @cmungall, reopening to tackle what ever happened to the indirect/direct notice in the neighborhood view.

**Comment by kltm** [comment]
Well, obvious answer: the neighborhood view is (now) just parents and children, so no need for further marking.
@cmungall Can you find an example where this is not true?

**Comment by kltm** [comment]
More talk with @cmungall , may actually be a difference here if the fixes did not make it up back up far enough:
getLineageShuntGraphJSON vs. (getNeighborsJSON and getNeighborsLimitedJSON)

**Comment by kltm** [comment]
Which...ties it back to: https://github.com/geneontology/amigo/issues/220, so we /want/ to see these relations, yes? So that brings us back to allowing these (for the "Planteome" use case), but marking them.

**Comment by ValWood** [comment]
Looking through tickets I opened (just because I landed in the AmiGIO tracker tonight). Is this still relevant? It does not bother me much because I never use or demonstrate this view....
